tmcprototype/CentralNode/CentralNode/CentralNode.py

```diff
@@ -511,6 +511,96 @@ def AssignResources(self, argin):
+
+        :param argin: The string in JSON format. The JSON contains following values:
+
+            subarrayID:
+                DevShort. Mandatory.
+
+            releaseALL:
+                Boolean(True or False). Mandatory. True when all the resources to be released from Subarray.
+
+            receptorIDList:
+                DevVarStringArray. Empty when releaseALL tag is True.
+
+            Example:
+                {
+                    "subarrayID": 1,
+                    "releaseALL": true
+                    "receptorIDList": []
+                }
+
+
+            Note: From Jive, enter input as:
+                {"subarrayID":1,"releaseALL":true,"receptorIDList":[]} without any space.
+
+            :return: argout: The string in JSON format. The JSON contains following values:
+
+                releaseALL:
+                    Boolean(True or False). If True, all the resources are successfully released from the Subarray.
+
+                receptorIDList:
+                    DevVarStringArray. If releaseALL is True, receptorIDList is empty. Else list returns resources
+                    (device names) that are noe released from the subarray.
+
+                Example:
+                    argout =
+                    {
+                        "ReleaseAll" : True,
+                        "receptorIDList" : []
+                    }
+        """
+
+        try:
+            jsonArgument = json.loads(argin)
+            subarrayID = jsonArgument['subarrayID']
+            subarrayProxy = self.subarray_FQDN_dict[subarrayID]
+            subarray_name = "SA" + str(subarrayID)
+            if jsonArgument['releaseALL'] == True:
+                self._command_success = subarrayProxy.command_inout(CONST.CMD_RELEASE_RESOURCES)
+                self._read_activity_message = CONST.STR_REL_RESOURCES
+                self.dev_logging(CONST.STR_REL_RESOURCES, int(tango.LogLevel.LOG_INFO))
+                if not self._command_success:
+                    for Dish_ID, Dish_Status in self._subarray_allocation.items():
+                        if Dish_Status == subarray_name:
+                            self._subarray_allocation[Dish_ID] = "NOT_ALLOCATED"
+                else:
+                    self._read_activity_message = CONST.STR_LIST_RES_NOT_REL \
+                                                  + self._command_success
+            else:
+                self._read_activity_message = CONST.STR_FALSE_TAG
+
+        except ValueError:
+            self.dev_logging(CONST.ERR_INVALID_JSON, int(tango.LogLevel.LOG_ERROR))
+            self._read_activity_message = CONST.ERR_INVALID_JSON
+        except KeyError:
+            self.dev_logging(CONST.ERR_JSON_KEY_NOT_FOUND, int(tango.LogLevel.LOG_ERROR))
+            self._read_activity_message = CONST.ERR_JSON_KEY_NOT_FOUND
+
+        argout = {
+            "ReleaseAll" : True,
+            "receptorIDList" : []
```

Comment:
If the API fails, the ReleaseALL tag should be False and Receptor id should have list of resources which are not released.

Reply:
Yes. We have changed the functionality as per suggestion.

Reply:
The commit id: [485e1cec5f7608502b3bafde2c932940c3532cfb](https://github.com/ska-telescope/tmc-prototype/pull/21/commits/485e1cec5f7608502b3bafde2c932940c3532cfb)